tests/test_authority.py

```diff
@@ -123,3 +128,95 @@ class MockResponse(object):
         finally:  # MUST NOT let the previous test changes affect other test cases
             Authority._domains_without_user_realm_discovery = set([])
 
+
+@patch("msal.authority.tenant_discovery", return_value={
+    "authorization_endpoint": "https://contoso.com/placeholder",
+    "token_endpoint": "https://contoso.com/placeholder",
+    })
+@patch("msal.authority.instance_discovery")
+@patch.object(msal.ClientApplication, "_get_instance_metadata", return_value=[])
+class TestMsalBehaviorsWithoutAndWithKnownAuthorityHosts(unittest.TestCase):
+    """Test cases use ClientApplication, which is a base class of both PCA and CCA"""
+
+    def test_by_default_a_known_to_microsoft_authority_should_skip_validation_but_still_use_instance_metadata(
+            self, instance_metadata, known_to_microsoft_validation, _):
+        app = msal.ClientApplication("id", authority="https://login.microsoftonline.com/common")
+        known_to_microsoft_validation.assert_not_called()
+        app.get_accounts()  # This could make an instance metadata call for authority aliases
```

Comment:
Is it expected in this case that `get_accounts()` does instance discovery? That is to say, `ClientApplication()` doesn't call `known_to_microsoft_validation` but `get_accounts()` does.

Reply:
The answer is two-fold.

* The "validating whether an input authority looks legit" and "getting some metadata for alias grouping" are two different things. This test case, as described in line 141, made a point that one does not necessarily turn off the other. In this case, `ClientApplication()` skips validation, but `get_accounts()` would like to know alias groups.
* The actual timing of when to send out the metadata call, is an implementation detail. Since MSAL Python delays that metadata call to `get_accounts()`, so this test case is written this way. Your MSAL's implementation does not have to call `get_accounts()`.

Last but not least, this entire PR has been shelved, because later we discovered some [new requirements for Azure Stack scenarios (internal link)](https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview?path=/AuthorityValidation/authority_validation_concepts_and_api.md&_a=preview&anchor=skip-authority-validation-and-alias-metadata-behaviors-without-a-set-of-predefined-authorities), so we proceeded with #496.